src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -436,6 +475,60 @@ private static byte[] ConvertEventData2Bytes(EventData input)
         private static EventData ConvertString2EventData(string input) 
             => ConvertBytes2EventData(Encoding.UTF8.GetBytes(input));
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "MessagingFactory")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "EventHubClient")]
+        internal static Uri GetEndpointFromEventHubClient(EventHubClient client)
+        {
+            var messagingFactoryProperty = typeof(EventHubClient).GetProperty("MessagingFactory", BindingFlags.Instance | BindingFlags.NonPublic);
+            if (messagingFactoryProperty == null)
+            {
+                throw new InvalidOperationException("Reflection call to obtain EventHubClient.MessagingFactory property info failed. Did a service bus package update occur?");
+            }
+
+            var messagingFactory = (MessagingFactory)messagingFactoryProperty.GetValue(client);
```

Comment:
>var messagingFactory = (MessagingFactory [](start = 12, length = 40)

nit:  since the property is internal, it may change the return type and then this would become a cast exception.  Suggested: 
    var messagingFactory = messagingFactoryProperty.GetValue(client) as MessagingFactory;
Then your nicer exception will still fire instead of the cast.